### ISSUE TITLE
GM 3.0 and 3.0 beta1 ignoring @noframes

### ISSUE THREAD

**Issue by Yonezpt** [greasemonkey/greasemonkey#2110]
After ~4 hours of trying to understand why my script started throwing errors in the past couple of days I finally discovered that the problem lied with the new Greasemonkey version and beta as well; The addon is ignoring the `@noframes` rule.

I don't know yet if this was by design or if it was overlooked and I don't see it mentioned in the changelog, but all scripts will now run in iframes despite having `@noframes` declared.

Sample script:

````javascript
// ==UserScript==
// @name        Test
// @namespace   Test
// @version     1
// @run-at      document-start
// @match       https://www.youtube.com/*
// @grant       none
// @noframes
// ==/UserScript==
if (self === top) {
    console.info('I am not an iframe');
} else if (self !== top) {
    console.info('I am an iframe');
}
````

Open https://www.youtube.com/watch?v=8pRJn77UeOM, for example, and observe the console.

![ss](https://cloud.githubusercontent.com/assets/8498057/6659789/43603814-cb7c-11e4-80b9-0d4a7e1b60e8.png)

The 3 iframe related strings show that the script is running inside iframes.

**Comment by janekptacijarabaci** [comment]
Confirmed.

script.noframes - undefined

https://github.com/greasemonkey/greasemonkey/blob/master/modules/ipcscript.js#L16

From:
```javascript
  this.namespace = aScript.namespace;
```

To:
```javascript
  this.namespace = aScript.namespace;
  this.noframes = aScript.noframes;
```

And maybe add:

https://github.com/greasemonkey/greasemonkey/blob/master/modules/ipcscript.js#L73

From:
```javascript
      'namespace': this.namespace,
```

To:
```javascript
      'namespace': this.namespace,
      'noframes': this.noframes,
```

**Comment by arantius** [comment]
This fix is now included in version 3.1beta1.  Please browse to [Greasemonkey's AMO page](https://github.com/Ventero/greasemonkey/commit/b8a09369482f87d0d9fa8afb39b5f46439eab0c2) and open "Development Channel" near the bottom to install the beta version.  Can you confirm the fix?  Thank you.

**Comment by Yonezpt** [comment]
I can confirm, the issue is fixed on the version you mentioned. Thanks for the quick response.
